pkg/template/core/starlark_value.go

```diff
@@ -55,37 +58,37 @@ func (e StarlarkValue) AsInt64() (int64, error) {
 	return 0, fmt.Errorf("expected starlark.Int")
 }
 
-func (e StarlarkValue) asInterface(val starlark.Value) interface{} {
+func (e StarlarkValue) asInterface(val starlark.Value) (interface{}, error) {
 	if obj, ok := val.(UnconvertableStarlarkValue); ok {
-		panic(obj.ConversionHint())
+		return nil, errors.New(obj.ConversionHint())
```

Comment:
i think we should have something like `fmt.Errorf("xxx: %s")` where `xxx` is some generic text that indicates that value is unconvertable.

Reply:
yup, makes sense.

fixed.